third-party/CMakeLists.txt

```diff
@@ -11,6 +11,12 @@ if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release)
 endif()
 
+if(CMAKE_BUILD_TYPE MATCHES Debug)
+  set(DEBUG 1)
+else()
+  set(DEBUG 0)
```

Comment:
Isn't this already set by now?

Reply:
`nvim` (`CMakeLists.txt`) and `NVIM_DEPS` (`third-party/CMakeLists.txt`) are two separate cmake projects with separate caches. Since `-DCMAKE_BUILD_TYPE` is never provided to `third-party/CMakeLists.txt`, `DEBUG` isn't set as well.

The only code that sets `CMAKE_BUILD_TYPE` in `NVIM_DEPS` so far is:

```cmake
# In Windows/MSVC CMAKE_BUILD_TYPE changes the paths/linking of the build
# recipes (libuv, msgpack), make sure it is set
if(NOT CMAKE_BUILD_TYPE)
  set(CMAKE_BUILD_TYPE Release)
endif()
```

Reply:
`DEBUG` is specific to unibilium, so this check should live in `third-party/cmake/BuildUnibilium.cmake`.